packages/config/src/load.ts

```diff
@@ -0,0 +1,155 @@
+  //
+  // for esm, before we can register loaders without requiring users to run node
+  // with --experimental-loader themselves, we have to do a hack here:
+  // write it to disk, load it with native Node ESM, then delete the file.
+  const fileBase = `${fileName}.timestamp-${Date.now()}`;
+  const fileNameTmp = `${fileBase}.mjs`;
+  const fileUrl = `${pathToFileURL(fileBase)}.mjs`;
+  await fs.writeFile(fileNameTmp, bundledCode);
+
+  try {
+    const configModule = await import(fileUrl);
+    validateConfigHasDefaultExport(configModule);
+
+    return configModule.default;
+  } finally {
+    await fs.unlink(fileNameTmp);
+  }
+}
+
+/**
+ * Pulls the user config from an optionally specified filepath.
+ *
+ * By default, this will search the current working directory.
+ *
+ * @param options options for loading the config file, defaults to `{}`. Can pass in `cwd` as a path `string` to override the current working directory.
+ * @returns Promise that resolves to the user config object.
+ */
+export async function loadConfig({ cwd = process.cwd() } = {}): Promise<
+  ConfigResult<FlatbreadConfig>
+> {
+  let configFileName: ConfigFileName;
+  const files = await fs.readdir(cwd);
+
+  const matchingFiles = files.filter((file) =>
+    FLATBREAD_CONFIG_FILE_REGEX.test(file)
+  ) as ConfigFileName[];
+
+  if (matchingFiles.length > 1) {
+    console.error(
+      colors.red(
+        `
+        Multiple config files found (${colors.gray(matchingFiles.join(', '))}).
+        
+        Please declare ${colors.bold('only')} one! 😅\n
+        `
+      ) + VALID_CONFIG_NAMES_MESSAGE
+    );
+    process.exit(1);
+  } else if (matchingFiles.length === 1) {
+    // Grab the config file name declared in the user's project root
+    configFileName = matchingFiles[0];
+  } else {
+    console.error(
+      colors.red(
+        `No config file found. Please declare one! 😅\n
+      `
+      ) + VALID_CONFIG_NAMES_MESSAGE
+    );
+    process.exit(1);
+  }
+
+  const configFilePath = path.join(cwd, configFileName);
+  const { code } = await bundleConfigFile(cwd, configFileName);
+  const config = await esmLoader(configFileName, code);
+
+  return {
+    filepath: configFilePath,
+    config: config,
+  };
+}
+
+/**
+ * Bundle the config file with esbuild and return the code.
+ * @param fileName config file name
+ * @returns
+ */
+async function bundleConfigFile(
+  cwd: string,
+  fileName: string
+): Promise<{ code: string }> {
+  const configBuild = await build({
+    absWorkingDir: cwd,
+    entryPoints: [fileName],
+    write: false,
+    platform: 'node',
+    bundle: true,
+    format: 'esm',
+    sourcemap: 'inline',
+    sourcesContent: false,
+    metafile: true,
+    banner: {
+      // Workaround for "Dynamic require of "os" is not supported" issue https://github.com/evanw/esbuild/issues/1921
+      js: "import { createRequire } from 'module';const require = createRequire(import.meta.url);",
```

Comment:
lol nice, I've ran into this same problem a couple times now

Reply:
yeah that was an annoying rabbit hole